Add configuration settings documentation for v1.0

benjaminwil opened the pull request:
What is the goal of this PR?
---

Provide brief descriptions of available configuration settings.


How do you manually test these changes? (if applicable)
---

Reviewers should read the documentation and check it against the implemented settings.

Noah-Silvera commented on README.md:
```diff
@@ -112,10 +106,111 @@ SuperGood::SolidusTaxjar.tap do |config|
+
+- `discount_calculator`: A discount calculator class. A safe default
+  calculator is provided, but may not be suitable for
+  stores. Check with your accountant how discounts need
```
I think the main concern here is that our discount calculator doesn't handle order level adjustments. 

Instead of asking users to talk to their accountants, maybe we should phrase it like this!

```
A discount calculator class. A default
calculator is provided, but may not be suitable for
stores. TaxJar requires discounts to be specified per
line item, so a custom calculator will be required to
support order level discounts and other promotions.
```

forkata replied:
```
A default calculator is provided, but may not be suitable for stores.
```

I think this statement generalizes a bit and for most stores might not be true. We shouldn't assume that the default calculator is not suitable for most stores, since promotions which rely on order level adjustments are _known_ to not work well with the built-in tax system as well so they are not widely used anymore.

I think as an extension whose main purpose is to provide tax calculations, we should probably phrase this in a way that mentions the exception to the rule, rather than start off with "this might not work for you".

```
A discount calculator class. A default calculator is provided which handles promotions which rely on line item and shipment adjustments. Since TaxJar requires discounts to be specified per line item, you will need to provide a custom version of this calculator in order to support promotions which create order level adjustments. Adjustments to an order made through the Solidus admin would also need to be handled through a custom implementation of this class.
```

Noah-Silvera commented on README.md:
```diff
@@ -112,10 +106,111 @@ SuperGood::SolidusTaxjar.tap do |config|
+
+  Default value: (See the source code.)
+
+- `job_queue`: A string or symbole that matches the name of a
```
symbol**é**

![image](https://user-images.githubusercontent.com/8933450/207169162-108cc1f1-6537-468b-aa1e-3b02143dd234.png)

Noah-Silvera commented on README.md:
```diff
@@ -112,10 +106,111 @@ SuperGood::SolidusTaxjar.tap do |config|
+  Default value: "Sales Tax"
+
+- `taxable_address_check`: A function that returns a boolean after checking
+  whether an address should be considered taxable.  By default this function
```
After [this PR](https://github.com/SuperGoodSoft/solidus_taxjar/pull/205), the default will be true unless you are in the united states and the address is **not** in a nexus region

benjaminwil replied:
Okay. I will leave as is, and if this gets merged first I can add a commit to that PR.

benjaminwil replied:
@forkata and I just double-checked. We've refactored the linked PR and we no longer change the `taxable_address_check`'s default return value.

Noah-Silvera reviewed (commented):
Thanks for kicking the documentation ticket off Benjamin!!

benjaminwil commented:
@forkata @Noah-Silvera 

Thanks for the review. I've addressed your feedback and added some more documentation:

- I've added other 1.0 features to the **Features** section.
- I added a new section, **Order-level Adjustments are Unsupported** to explain to developers what they need to do to support order-level adjustments should they need to.
